src/core/stream.js

```diff
@@ -735,6 +735,19 @@ var PredictorStream = (function PredictorStreamClosure() {
         buffer[pos] = buffer[pos - colors] + rawBytes[i];
         pos++;
       }
+    } else if (bits === 16) {
```

Comment:
In the issue you have a test file for this. I don't know what its license is, but are we free to include that in the test suite? If so, please add it as a regular test case (see https://github.com/mozilla/pdf.js/pull/8917/files#diff-441ddcbb46641da096fad809396ea452 for an example: add the file, a `.gitignore` entry and a test manifest entry). If not, please add it as a linked test case (see https://github.com/mozilla/pdf.js/pull/8913/files#diff-9506708851a1dcc6c972f0831003f3d5 for an example: add the file in a `.link` file and add a test manifest entry). This is to make sure we don't regress this in the future.

Reply:
Re squashing commits, this will take some time. I agree that you want to keep your commit history clean of "oops"-type commits. If I understand it properly, this isn't possible from the website. I'll first try from my IDE, and if that doesn't do it, I'll have to install a command line client :-(

Reply:
Surprisingly this didn't take hours or days. Thanks for the link and for your help.